cmd/drone-buildx-gcr/main.go

```diff
@@ -1,63 +1,159 @@
 package main
 
 import (
+	"context"
 	"encoding/base64"
+	"log"
 	"os"
 	"path"
+	"strconv"
 	"strings"
 
 	docker "github.com/drone-plugins/drone-buildx"
+	"github.com/drone-plugins/drone-buildx-gcr/internal/gcp"
 	"github.com/joho/godotenv"
+	"github.com/sirupsen/logrus"
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/google"
 )
 
-// gcr default username
-const username = "_json_key"
+type Config struct {
+	Repo             string
+	Registry         string
+	Password         string
+	WorkloadIdentity bool
+	Username         string
+	AccessToken      string
+}
 
-func main() {
-	// Load env-file if it exists first
+type staticTokenSource struct {
+	token *oauth2.Token
+}
+
+func (s *staticTokenSource) Token() (*oauth2.Token, error) {
+	return s.token, nil
+}
```

Comment:
Can this be imported from either GCP package or tokenutil instead of duplicating?

Reply:
removed repeating/unused blocks